experiments/axon_geometry/scripts/generate_trace_data.py

```diff
@@ -55,7 +60,7 @@ def generate_brain_trace_data(brain: str):
                 # evaluate segment length (in um)
                 seg_length = u_um[-1] - u_um[0]
                 # resample points at 1um
-                uu = np.arange(u_um[0], u_um[-1] + 0.9, 1)
+                uu = np.arange(u_um[0], u_um[-1] + 0.9, spacing)
```

Comment:
Good point. Here's my proposal: it's most important to _not_ sample beyond the end of the segment. Since u_um[-1] will typically be a non-integer, the best thing is just remove the added margin at the end.

For your second comment, you say "twice the spacing" but there's no "twice" factor right? e.g. a segment that is 15um long will be sampled twice with a spacing of 14um (one at 0, one at 14).

Reply:
Ok! I agree:
```suggestion
                uu = np.arange(u_um[0], u_um[-1], spacing)
```

Yea, idk why I said that...

Reply:
I'm just going to remove the + 0.9 altogether. Because say u_um[-1] is 1.9, then the resulting sampling with a spacing of 1 would be [0,1,2] but 2 falls beyond the length of 1.9.